src/corehost/cli/bdl_file_entry.cpp

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifdef FEATURE_APPHOST
+
+#include "bdl_processor.h"
+#include "pal.h"
+#include "error_codes.h"
+#include "trace.h"
+#include "utils.h"
+
+bool file_entry_t::is_valid()
+{
+	return data.offset > 0 && data.size > 0 && 
+		   (file_type_t)data.type < file_type_t::END &&
+		   data.path_length > 0 && data.path_length <= PATH_MAX;
+}
+
+file_entry_t* file_entry_t::read(FILE* bundle)
+{
+	file_entry_t* entry = new file_entry_t();
+
+	// First read the fixed-sized portion of file-entry
+	bdl_processor_t::read(&entry->data, sizeof(entry->data), bundle);
+	if (!entry->is_valid())
+	{
+		trace::error(_X("Invalid FileEntry"));
```

Comment:
Please make the errors more descriptive. Consider that they are visible be end users. Not only the developers, but the actual users of the apps, which may not know anything about bundles. The error should start with some kind of humanly readable part and then we can include the technical details.

Maybe something like:
> Failed to open the application bundle. It's possibly corrupted.
> Invalid file entry found.

Please ask some native speaker to polish this...